src/solver/utils/ortools_utils.cpp

```diff
@@ -348,178 +344,66 @@ void ORTOOLS_LibererProbleme(MPSolver* solver)
+            result.push_back((std::string)ToString(solverType));
     }
 
     return result;
 }
 
-std::list<OrtoolsSolver> OrtoolsUtils::getAvailableOrtoolsSolver()
-{
-    std::list<OrtoolsSolver> result;
-
-    std::list<OrtoolsSolver> solverList = Enum::enumList<Antares::Data::OrtoolsSolver>();
 
-    for (OrtoolsSolver solver : solverList)
-    {
-        if (isOrtoolsSolverAvailable(solver))
-        {
-            result.push_back(solver);
-        }
-    }
-
-    return result;
-}
-
-std::list<std::string> OrtoolsUtils::getAvailableOrtoolsSolverName()
+std::list<std::string> getAvailableOrtoolsSolverName()
 {
     std::list<std::string> result;
 
-    std::list<OrtoolsSolver> solverList = getAvailableOrtoolsSolver();
-
-    for (OrtoolsSolver solver : solverList)
+    for (const auto solverName : OrtoolsUtils::solverMap)
     {
-        result.push_back(Enum::toString(solver));
-    }
-
-    return result;
-}
-
-MPSolver::OptimizationProblemType OrtoolsUtils::getLinearOptimProblemType(
-  const OrtoolsSolver& ortoolsSolver)
-{
-    // TODO JMK : see how we can get optimization problem type with current ortools RTE branch
-    // (can't use enum because of compile switch) MPSolver::OptimizationProblemType result =
-    // _solverLinearProblemOptimMap.at(ortoolsSolver)
+        MPSolver::OptimizationProblemType solverType;
+        MPSolver::ParseSolverType(solverName.second.first, &solverType);
 
-    // For now we are using string to get optimization problem type
-    MPSolver::OptimizationProblemType result
-      = MPSolver::OptimizationProblemType::GLOP_LINEAR_PROGRAMMING;
-
-    if (!MPSolver::ParseSolverType(_solverLinearProblemOptimStringMap.at(ortoolsSolver), &result))
-    {
-        throw AssertionError("Unsupported Linear OrtoolsSolver for solver "
-                             + Enum::toString(ortoolsSolver));
+        if (MPSolver::SupportsProblemType(solverType))
+            result.push_back(solverName.first);
     }
 
     return result;
 }
 
-MPSolver::OptimizationProblemType OrtoolsUtils::getMixedIntegerOptimProblemType(
-  const OrtoolsSolver& ortoolsSolver)
-{
-    // TODO JMK : see how we can get optimization problem type with current ortools RTE branch
-    // (can't use enum because of compile switch) MPSolver::OptimizationProblemType result =
-    // _solverMixedIntegerProblemOptimMap.at(ortoolsSolver)
 
-    // For now we are using string to get optimization problem type
-    MPSolver::OptimizationProblemType result
-      = MPSolver::OptimizationProblemType::GLOP_LINEAR_PROGRAMMING;
+MPSolver* factory(const Antares::Optimization::PROBLEME_SIMPLEXE_NOMME *probleme, std::string solverName)
+{
+    std::list<std::string> solverList = getAvailableOrtoolsSolverName();
 
-    if (!MPSolver::ParseSolverType(_solverMixedIntegerProblemOptimStringMap[ortoolsSolver],
-                                   &result))
+    auto it = std::find(solverList.begin(), solverList.end(), solverName);
+    if (it != solverList.end())
```

Comment:
There's no need to loop in this list.

`CreateSolver` returns `nullptr` in case the solver isn't supported.

You can do the following

1. Call `CreateSolver`
2. If equal to `nullptr`, return an error
3. Return the newly allocated object otherwise